src/app/qgisapp.cpp

```diff
@@ -13217,23 +13224,36 @@ Qgs3DMapCanvas *QgisApp::createNewMapCanvas3D( const QString &name )
     QgsSettings settings;
 
     Qgs3DMapSettings *map = new Qgs3DMapSettings;
-    if ( !prj->crs3D().isGeographic() )
+    if ( sceneMode == Qgis::SceneMode::Globe )
     {
-      map->setCrs( prj->crs3D() );
+      // Geocentric CRS based on WGS 84
+      map->setCrs( QgsCoordinateReferenceSystem( QStringLiteral( "EPSG:4978" ) ) );
```

Comment:
```suggestion
      // Geocentric CRS based on WGS 84
      // TODO: dynamically build geocentric CRS based on project's ellipsoid to permit non-earth globes
      map->setCrs( QgsCoordinateReferenceSystem( QStringLiteral( "EPSG:4978" ) ) );
```

Reply:
https://github.com/qgis/QGIS/pull/61268 relates -- with that we'll be able to just directly build the map crs from the project's CRS, using:

    map->setCrs( QgsProject::instance()->crs().toGeocentricCrs() );

ensuring that non-earth globes will work